src/pudl/io_managers.py

```diff
@@ -855,12 +855,13 @@ def load_input(self, context: InputContext) -> pd.DataFrame:
             context: dagster keyword that provides access output information like asset
                 name.
         """
-        # TODO (daz): this is hard-coded to FERC1, though this is nominally for all FERC datasets.
-        ferc1_settings = context.resources.dataset_settings.ferc1
+        ferc_settings = context.resources.dataset_settings[
+            self.db_name.replace("_xbrl", "")
+        ]
```

Comment:
the `dataset_settings` here is a pydantic object so you can't access this directly like a dictionary. but you can do this:
```suggestion
        ferc_settings = context.resources.dataset_settings.get_datasets()[
            self.db_name.replace("_xbrl", "")
        ]
```

Reply:
done!